ci/deploy.sh

```diff
@@ -0,0 +1,96 @@
+  echo "  -a, --app                       App Name."
+  echo "  -e, --env                       Application environment variables."
+  echo "  -ig, --ingress                  Ingress type. Default is Traefik."
+  echo "  --endpoint                      Ingress IP address."
+  echo "  -i, --image                     The image that should be used with the application."
+  echo "  --registry-secret               A name of a Secret with docker credentials. This secret must be created in the same namespace of the pool."
+  echo "  --ketch-yaml                    The path to the ketch.yaml file."
+  echo "  --procfile	                  The path to Procfile. If not set, ketch will use the entrypoint and cmd from the image."
+  echo "  --skip-resource-creation        If set, ketch will NOT create app and pool for the deployment. Useful when resources already exist."
+  exit 1
+}
+
+# parse params and set variables with custom user inputs
+while [[ "$#" > 0 ]]; do case $1 in
+    -t|--ketch-tag) KETCH_TAG="$2"; shift;shift;;
+    -o|--pool) POOL="$2"; shift;shift;;
+    -ig|--ingress) INGRESS_TYPE="$2"; shift;shift;;
+    --endpoint) INGRESS_ENDPOINT="$2"; shift;shift;;
+    -a|--app) APP_NAME="$2"; shift;shift;;
+    -e|--env) APP_ENV="$2"; shift;shift;;
+    -i|--image) DOCKER_IMAGE="$2"; shift;shift;;
+    --registry-secret) REG_SECRET="$2"; shift;shift;;
+    --ketch-yaml) KETCH_YAML="$2"; shift;shift;;
+    --procfile) PROCFILE="$2"; shift;shift;;
+    --skip-resource-creation) RESOURCE_CREATION=false; shift;shift;;
+    *) usage "Unknown parameter passed: $1"; shift; shift;;
+esac; done
+
+
+# validate params
+if [ -z "$APP_NAME" ]; then usage "App Name required"; fi;
+if [ -z "$DOCKER_IMAGE" ]; then usage "Image for the app is required"; fi;
+
+# set defaults if not set by user
+if [ -z "$KETCH_TAG" ]; then 
+    KETCH_TAG=$(curl -s https://api.github.com/repos/shipa-corp/ketch/releases/latest | grep -Eo '"tag_name":.*[^\\]",' | head -n 1 | sed 's/[," ]//g' | cut -d ':' -f 2)
+fi
+
+# Install ketch binary at /usr/local/bin default location
+ curl -s https://raw.githubusercontent.com/shipa-corp/ketch/main/install.sh | TAG="${KETCH_TAG}" bash
+
+# Install Ketch controller
+kubectl apply -f https://github.com/shipa-corp/ketch/releases/download/"${KETCH_TAG}"/ketch-controller.yaml
+
+if [ $RESOURCE_CREATION ] ; then
+    # validate addtional required params
+    if [ -z "$POOL" ]; then usage "Pool name required"; fi;
+    if [ -z "$INGRESS_ENDPOINT" ]; then usage "Ingress endpoint required"; fi;
+
+    # Add a pool with ingress Traefik (default), replace ingress endpoint address by your ingress IP address
+    echo "creating pool for deployment ..."
+    ketch pool add "${POOL}"  --ingress-service-endpoint "${INGRESS_ENDPOINT}" --ingress-type "${INGRESS_TYPE}"
+
+    # Create app
+    echo "creating app for deployment ..."
+    ketch app create "${APP_NAME}" --pool "${POOL}"  --env "${APP_ENV}" --registry-secret "${REG_SECRET}"
```

Comment:
Also check for APP_NAME and print usage message, just like POOL and INGRESS_ENDPOINT?

Reply:
We already do it [here](https://github.com/shipa-corp/ketch/pull/32/files#diff-d498a12aa774941c9553350f6bc684a7bb2a70d6b6a2c1fb0619dd90907e8d4dR68)

These checks ( POOL, INGRESS_ENDPOINT) are under the `if` block as they might not be required if --skip-resource-creation is not set.

Reply:
ah, missed --skip-resource-creation flag. thanks.